src/js/conntest.js

```diff
@@ -7,100 +7,72 @@
  */
 'use strict';
 
-addTest(testSuiteName.CONNECTIVITY, testCaseName.UDPCONNECTIVITY,
-    udpConnectivityTest);
-addTest(testSuiteName.CONNECTIVITY, testCaseName.TCPCONNECTIVITY,
-    tcpConnectivityTest);
-addTest(testSuiteName.CONNECTIVITY, testCaseName.IPV6ENABLED,
-    hasIpv6CandidatesTest);
+addTest(testSuiteName.CONNECTIVITY, testCaseName.RELAYCONNECTIVITY,
+    relayConnectivityTest);
+addTest(testSuiteName.CONNECTIVITY, testCaseName.REFLEXIVECONNECTIVITY,
+    reflexiveConnectivityTest);
+addTest(testSuiteName.CONNECTIVITY, testCaseName.HOSTCONNECTIVITY,
+    hostConnectivityTest);
 
-// Test whether it can connect via UDP to a TURN server
-// Get a TURN config, and try to get a relay candidate using UDP.
-function udpConnectivityTest() {
-  Call.asyncCreateTurnConfig(
-      function(config) {
-        filterConfig(config, 'udp');
-        gatherCandidates(config, null, Call.isRelay);
-      },
-      reportFatal);
-}
+var timeout = null;
 
-// Test whether it can connect via TCP to a TURN server
-// Get a TURN config, and try to get a relay candidate using TCP.
-function tcpConnectivityTest() {
+// Set up a datachannel between two peers through a relay
+// and verify data can be transmitted and received
+// (packets travel through the public internet)
+function relayConnectivityTest() {
   Call.asyncCreateTurnConfig(
-      function(config) {
-        filterConfig(config, 'tcp');
-        gatherCandidates(config, null, Call.isRelay);
-      },
-      reportFatal);
-}
-
-// Test whether it is IPv6 enabled (TODO: test IPv6 to a destination).
-// Turn on IPv6, and try to get an IPv6 host candidate.
-function hasIpv6CandidatesTest() {
-  var params = {optional: [{googIPv6: true}]};
-  gatherCandidates(null, params, Call.isIpv6);
+    runConnectivityTest.bind(null, Call.isRelay), reportFatal);
 }
 
-// Filter the RTCConfiguration |config| to only contain URLs with the
-// specified transport protocol |protocol|. If no turn transport is
-// specified it is added with the requested protocol.
-function filterConfig(config, protocol) {
-  var transport = 'transport=' + protocol;
-  for (var i = 0; i < config.iceServers.length; ++i) {
-    var iceServer = config.iceServers[i];
-    var newUrls = [];
-    for (var j = 0; j < iceServer.urls.length; ++j) {
-      var uri = iceServer.urls[j];
-      if (uri.indexOf(transport) !== -1) {
-        newUrls.push(uri);
-      } else if (uri.indexOf('?transport=') === -1 && uri.startsWith('turn')) {
-        newUrls.push(uri + '?' + transport);
+// Set up a datachannel between two peers through a public IP address
+// and verify data can be transmitted and received
+// (packets should stay on the link if behind a router doing NAT)
+function reflexiveConnectivityTest() {
+  runConnectivityTest(Call.isReflexive, {
```

Comment:
I think we should use the turn servers provided by https://github.com/webrtc/testrtc/blob/master/src/js/call.js#L160. Do note that you will get a list of  TURN servers, however they do STUN as well, just change the URL to stun when stun is required:
`turn:23.251.129.26:3479 > stun:23.251.129.26:3479`